lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -65,16 +66,23 @@ def obfuscate_hostname?(host)
       end
 
       def fqdn(host)
-        return host.fqdn unless obfuscate_hostname?(host)
-
-        fact_value(host, 'insights_client::hostname') || obfuscate_fqdn(host.fqdn)
+        if obfuscate_hostname?(host)
+          if Setting[:insights_minimal_data_collection]
```

Comment:
This may work but it's confusing to me. The logic should be

1. If insights_minimal_data_collection, always obfuscate
2. If not, obfuscate as per the logic in `obfuscate_hostname?(host)`

Having logic both in `obfuscate_hostname?(host)` and `fqdn(host)` is hard to read and think about. I think `obfuscate_hostname?` should _only_ have the responsibility of checking with the non-minimal logic (in other words, don't check the setting there.) And then add back a guard clause here in `fqdn` that checks the setting.

Reply:
Fixed - reverted the methods, I was overthinking things when I made it. Tested all scenarios and they still work and unit tests are all passing.